core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -100,8 +101,8 @@ object ConsoleConsumer extends Logging {
         consumer.receive()
       } catch {
         case e: Throwable => {
-          error("Error processing message, stopping consumer: ", e)
-          consumer.stop()
+          error("Error processing message, terminating consumer process: ", e)
+          // Consumer will be stopped in shutdown hook
```

Comment:
Actually after talking with @hachikuji again I feel we still need both ```BaseConsumer.close()``` and ```BaseConsumer.cleanup()``` which will be implemented as wakeup and close in new consumer respectively. But it is safe to remove the close() call in catch exception but leave it only in shutdown hook since as you mentioned it is already out of poll already.